samples/bot-calling-meeting/csharp/Source/CallingBotSample/Services/BotFramework/ConnectorClientFactory.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Net.Http;
+using CallingBotSample.Options;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace CallingBotSample.Services.BotFramework
+{
+    /// <inheritdoc/>
+    public class ConnectorClientFactory : IConnectorClientFactory
+    {
+        private readonly BotOptions botOptions;
+        private readonly ConcurrentDictionary<string, ConnectorClient> connectorClients = new ConcurrentDictionary<string, ConnectorClient>();
+        private readonly ILogger<ConnectorClientFactory> logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConnectorClientFactory "/> class.
+        /// </summary>
+        /// <param name="logger">Logger.</param>
+        public ConnectorClientFactory(IOptions<BotOptions> botOptions, ILogger<ConnectorClientFactory> logger)
+        {
+            this.botOptions = botOptions.Value;
+            this.logger = logger;
+        }
+
+        /// <inheritdoc/>
+        public ConnectorClient CreateConnectorClient()
+        {
+            return CreateConnectorClient(
```

Comment:
Ah, we're using the global endpoint. Note here that this will only work in public cloud, not gov clouds.

Reply:
Do we really never get a specific service URL from any incoming event?

Reply:
Fair point. Let's note that and add a link to the doc page describing the global service URLs.